src/main/java/com/example/bookstoreapp/controller/CategoryController.java

```diff
@@ -0,0 +1,69 @@
+
+import com.example.bookstoreapp.dto.BookDto;
+import com.example.bookstoreapp.dto.CategoryDto;
+import com.example.bookstoreapp.dto.CreateCategoryDto;
+import com.example.bookstoreapp.servive.CategoryService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Categories management", description = "Endpoints for management categories")
+@RestController
+@RequiredArgsConstructor
+@PreAuthorize("hasRole('ADMIN')")
+@RequestMapping("/categories")
+public class CategoryController {
+    private final CategoryService categoryService;
+
+    @PostMapping
+    @Operation(summary = "Create a new category", description = "Create a new category")
+    @ResponseStatus(HttpStatus.CREATED)
+    CategoryDto createCategory(@RequestBody CreateCategoryDto createCategoryDto) {
+        return categoryService.save(createCategoryDto);
+    }
+
+    @GetMapping
+    @Operation(summary = "Get all categories", description = "Get all categories")
+    public List<CategoryDto> getAll(Pageable pageable) {
+        return categoryService.findAll(pageable);
+    }
+
+    @GetMapping("/{id}")
+    @Operation(summary = "Get category by ID", description = "Get category by ID")
+    public CategoryDto getCategoryById(@PathVariable Long id) {
+        return categoryService.findById(id);
+    }
+
+    @PutMapping("/{id}")
+    @Operation(summary = "Update category", description = "Update category")
+    public CategoryDto updateCategory(@PathVariable Long id, @RequestBody CategoryDto categoryDto) {
+        return categoryService.updateCategory(id, categoryDto);
+    }
+
+    @DeleteMapping("/{id}")
+    @Operation(summary = "Delete category", description = "Delete category")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void deleteCategory(@PathVariable Long id) {
+        categoryService.deleteById(id);
+    }
+
+    @GetMapping("/{id}/books")
+    @Operation(summary = "Get books by category ID", description = "Get books by category ID")
+    public List<BookDto> getBooksByCategoryId(@PathVariable Long id, Pageable pageable) {
+        return categoryService.getBooksByCategoryId(id, pageable);
+    }
```

Comment:
What is the difference between this and 'getBooksByCategory' method in BookController?

Reply:
None. My mistake